open_spiel/python/examples/lewis_signaling_dqn.py

```diff
@@ -0,0 +1,255 @@
+        except ValueError:
+            raise ValueError(
+                'There should be {} (states * actions) elements in payoff. Found {} elements'
+                .format(num_states * num_states, len(payoffs_list)))
+
+    env_configs = {
+        "num_states": num_states,
+        "num_messages": num_messages,
+        "payoffs": payoffs_str
+    }
+
+    env = rl_environment.Environment(game, **env_configs)
+    state_size = env.observation_spec()["info_state"][0]
+    num_actions = env.action_spec()["num_actions"]
+    replay_buffer_capacity = int(1e4)
+
+    # Results to store
+    num_runs = FLAGS.num_runs
+    training_episodes = FLAGS.num_episodes
+    log_interval = FLAGS.log_interval
+    rewards = np.zeros((num_runs, training_episodes // log_interval))
+    opts = np.zeros((num_runs, training_episodes // log_interval))
+    converge_point = np.zeros((num_states, num_states))
+    percent_opt = 0
+
+    # Repeat the experiment num_runs times
+    for i in range(num_runs):
+        with tf.Session() as sess:
+            agents = [
+                dqn.DQN(sess,
+                        player_id=idx,
+                        state_representation_size=state_size,
+                        num_actions=num_actions,
+                        learning_rate=FLAGS.step_size,
+                        replay_buffer_capacity=replay_buffer_capacity,
+                        epsilon_start=FLAGS.eps_init,
+                        epsilon_end=FLAGS.eps_final,
+                        epsilon_decay_duration=FLAGS.eps_decay_steps * 2)
+                for idx in range(num_players)
+            ]
+
+            # 1. Train the agents
+            for cur_episode in range(training_episodes):
+                time_step = env.reset()
+                # Find cur_state for logging. See lewis_signaling.cc for info_state details
+                cur_state = time_step.observations["info_state"][0][3:].index(1)
+                while not time_step.last():
+                    player_id = time_step.observations["current_player"]
+                    agent_output = agents[player_id].step(time_step)
+                    time_step = env.step([agent_output.action])
+
+                # Episode is over, step all agents with final info state.
+                for agent in agents:
+                    agent.step(time_step)
+
+                # Store rewards
+                reward = time_step.rewards[0]
+                max_reward = payoffs[cur_state].max()
+                cur_idx = (i, cur_episode // log_interval)
+                rewards[cur_idx] += reward / log_interval
+                opts[cur_idx] += np.isclose(reward, max_reward) / log_interval
+
+            base_info_state0 = [1.0, 0.0, 0.0] + [0.0] * num_states
+            base_info_state1 = [0.0, 1.0, 0.0] + [0.0] * num_states
+
+            for s in range(num_states):
+                info_state0 = copy.deepcopy(base_info_state0)
```

Comment:
Not sure if you need a copy! why not passig base_info_state0 to agents[0]._epsilon_greedy?